### PR TITLE
Polish billing UI

### PR DESCRIPTION
**Pull request by ryanjduffy**
## Summary

* Adds a stepped series of views for billing including a screen for express consent before collecting payment method
* Adds a `noTitle` optional config to `Settings` interface so the billing views can override the title per-step
* Includes a feature flag (off by default) to show trial duration

## Notes
Didn't record a video or replay since it involves capturing credit card info but here are a bunch of screenshots :)

Cancelled subscription with payment info
image

Cancelled subscription without payment info (there's a backend bug here on reporting the correct seat count)
image

Beta subscription with payment info
image

Beta subscription without payment info
image

Payment Consent
image

Add Payment Method
image

Confirm Payment Method
image

New Team With Trial (and feature flag to display trial info off)
image

### CODE REVIEW

**Comment by ryanjduffy** [comment]
One bug to note that I didn't have an obvious fix for is that clicking Billing when you're on a sub-view (e.g. entering card details) doesn't take you back to the first view. You can get there by clicking another menu or closing the view.

The challenge is telling `SettingsModal` to forcibly re-render the billing view or include a method passed by `SettingsModal` to the view to know when it was clicked. Neither are hard but a bit messy so I'd like to defer a bit on them.

**Review by jaril — APPROVED** [review]
Looks really good!

One last nit — it looks like there's a scrollbar to the right

<details>
<summary>
Screenshot</summary


</details>

File: src/ui/components/shared/WorkspaceSettingsModal/WorkspaceSubscription.tsx [hunk 1]
```diff
@@ -559,54 +570,159 @@ function AddPaymentMethod({ onDone, workspaceId }: { onDone: () => void; workspa
...
+        Terms of service and cancellation policy
+      </a>
+    </section>
+  );
+}
+
+function Confirmation({ subscription }: { subscription: Subscription }) {
+  return (
+    <section>
+      {/* <div className="h-36 mb-6 rounded-lg bg-blue-100" /> */}
```
[the remark this answers is not in the document]
Comment (ryanjduffy) [reply src/ui/components/shared/WorkspaceSettingsModal/WorkspaceSubscription.tsx:713]: Yeah. It's a placeholder for some future illustration.

**Comment by ryanjduffy** [comment]
Thanks. I was messing with always showing the scroller to see if it helped discoverability on the form and forgot to remove it. WIll do now.

**Pull request merged**